sdk/src/main/java/com/mapbox/maps/renderer/egl/EGLCore.kt

```diff
@@ -176,16 +192,60 @@ internal class EGLCore(
     return EGL10.EGL_SUCCESS
   }
 
-  /**
-   * Checks for EGL errors.
-   */
-  private fun checkEglErrorNoException(msg: String): Boolean {
+  internal fun addRendererStateListener(listener: RendererSetupErrorListener) {
+    rendererSetupErrorListenerSet.add(listener)
+    if (accumulatedRendererErrorList.isNotEmpty()) {
+      val errorListCopy = LinkedList(accumulatedRendererErrorList)
```

Comment:
Can you just take iterator() and clear afterwards instead of copy?

Reply:
did hide this under the lock, making a copy then is at least more clear for me to understand than using iterator for thread-safe collection.

Reply:
actually revisited and did as you said